✨ feat: Add support for Okta Authentication

JamieStivala opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change

<!-- Thank you for your Pull Request. Please provide a description above. -->
Added support for Okta authentication

#### 📝 补充信息 | Additional Information

<!-- Add any other context about the Pull Request here. -->

## Summary by Sourcery

Add support for Okta authentication by registering Okta environment variables for type declarations, validation, and configuration, emit deprecation warnings for them, update sign-in callback URL logic, and include tests for the new warnings.

New Features:
- Include OKTA_CLIENT_ID, OKTA_CLIENT_SECRET, and OKTA_ISSUER in environment type declarations, validation schema, and config output
- Emit deprecation warnings for Okta environment variables when they are set

Enhancements:
- Change next-auth sign-in callbackUrl default fallback from '/' to an empty string

Tests:
- Add unit test to verify deprecation warnings for Okta environment variables

actions-user committed:
Add Okta support to auth config and tests

arvinxx reviewed (changes requested):
the implement is not correctly , please check other pr like https://github.com/lobehub/lobe-chat/pull/7342. And add relative documents

JamieStivala committed:
Added documentation

JamieStivala committed:
Removed deprecated env variables

JamieStivala committed:
Added Okta as SSO Provider

JamieStivala commented:
Thank you for your reply. I have just added the necessary documentation (in English, sorry I am not able to translate to CH) as well as the provider information. 

I also removed the deprecated env variables as per PR #7342.

JamieStivala committed:
Removed Okta Test

JamieStivala committed:
Reverted a micro-change which was changed during some testing (back to original)

JamieStivala committed:
Added Okta to SSO providers list

JamieStivala committed:
Merge branch 'main' into feat/okta

arvinxx reviewed (approved):
LGTM. I will add Chinese docs for you. Thanks for your contribution!
